packages/core/src/external/analytics/posthog.ts

```diff
@@ -16,11 +16,12 @@ export interface EventMessageV1 extends IdentifyMessageV1 {
   timestamp?: Date;
 }
 
-const postApiKey = Config.getPostHogApiKey();
+const defaultPostHogApiKey = Config.getPostHogApiKey();
 
-export const analytics = (params: EventMessageV1) => {
-  if (postApiKey) {
-    const posthog = new PostHog(postApiKey);
+export const analytics = (params: EventMessageV1, postApiKey?: string) => {
+  const apiKey = postApiKey ?? defaultPostHogApiKey;
+  if (apiKey) {
```

Comment:
nit: since we're touching this, could make this return if no api key available, instead of having the main flow inside an if

NABD, though